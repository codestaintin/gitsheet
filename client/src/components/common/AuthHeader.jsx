import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const AuthHeader = ({ logout }) => (
  <Fragment>
    <nav
      className="navbar navbar-light"
      style={{ backgroundColor: '#3D4E81' }}
    >
      <ul className="navbar-nav">
        <li className="nav-item">
          <Link className="nav-link link" to="/">Git Sheet</Link>
        </li>
      </ul>

      <ul className="nav justify-content-end">
        <li className="nav-item">
          <Link
            to="#"
            className="nav-link link"
          >
            <button
              className="btn btn-outline-light"
              type="button"
              onClick={logout}
            >
              Log Out
            </button>
          </Link>
        </li>
      </ul>
    </nav>
  </Fragment>
);

AuthHeader.propTypes = {
  logout: PropTypes.func.isRequired
};

export default AuthHeader;