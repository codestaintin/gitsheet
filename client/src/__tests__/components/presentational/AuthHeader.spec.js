import React from 'react';
import { shallow } from 'enzyme';
import AuthHeader from '../../../components/common/AuthHeader';

describe('<AuthHeader>', () => {
  const props = {
    logout: jest.fn(() => {}),
    sidebarOpen: true
  };
  it('should render the AuthHeader component', () => {
    const tree = shallow(<AuthHeader {...props} />);
    expect(tree).toMatchSnapshot();
  });
});