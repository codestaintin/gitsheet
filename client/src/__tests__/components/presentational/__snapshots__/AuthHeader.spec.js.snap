// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthHeader> should render the AuthHeader component 1`] = `
<Fragment>
  <nav
    className="navbar navbar-light"
    style={
      Object {
        "backgroundColor": "#3D4E81",
        "zIndex": 0,
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link link"
          replace={false}
          to="/"
        >
          Git Sheet
        </Link>
      </li>
    </ul>
    <ul
      className="nav justify-content-end"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link link"
          replace={false}
          to="#"
        >
          <button
            className="btn btn-outline-light"
            onClick={[MockFunction]}
            type="button"
          >
            Log Out
          </button>
        </Link>
      </li>
    </ul>
  </nav>
</Fragment>
`;
